src/scss/settings/_buttons.scss

```diff
@@ -81,19 +86,57 @@ $buttonTypes: (
   high-contrast: (
     default: (
       background: punch,
-      textColor: air
+      textColor: air,
+      shadow: (
+        kind: normal
+      )
     ),
     hover: (
-      background: serious
+      background: serious,
+      shadow: (
+        kind: medium
+      )
+    ),
+    active: (
+      background: serious,
+      shadow: (
+        kind: none
+      )
+    ),
+    disabled: (
+      background: (color: punch, opacity: .2),
+      textColor: (color: serious, opacity: .2),
+      shadow: (
+        kind: none
+      )
     )
   ),
   high-contrast-reversed: (
     default: (
-      background: (color: air, opacity: .9),
-      textColor: punch
+      background: (color: air),
+      textColor: punch,
+      shadow: (
```

Comment:
This can be inline (`shadow: ( kind: normal ) )` and in another occurrences of this case (L124, L137, etc.)